### ISSUE TITLE
[feature request] Allow selection of audio track/language

### ISSUE THREAD

**Issue by chemeris** [Rafostar/gnome-shell-extension-cast-to-tv#117]
Great app! It lacks an important feature for international users though - an ability to select audio stream/language in a video file. Most movies I'm watching are coming with multiple audio tracks - original audio, and a selection of different translations.

Right now Cast to TV automatically plays the first audio track and doesn't allow me to select a different audio track and makes it quite useless to me :(

**Comment by Rafostar** [comment]
`Default Media Receiver` does not support audio track selection from single video file. Such function would require writing a new receiver app for Chromecast based on CAF API.

As a workaround I recommend manually re-merging selected tracks into a new video file (without transcoding). This can be done with `ffmpeg` or from GUI with `mkvtoolnix`.

**Comment by chemeris** [comment]
Yes, I'm aware of the re-merging method and that's what some other Linux Chromecast tools are doing under the hood - either into a temp file or even on the fly. So there is no need to do that manually with those other tools - they hide the re-merging from the user. Might be a relatively easy way to go for this tool as well?

**Comment by Rafostar** [comment]
@chemeris 
> either into a temp file

Remerging into temp file by extension is less than ideal option. If single video for example has 5GB size, then we would need to read 5GB of data from HDD, save almost 5GB (only a little smaller due to missing few audio tracks), then cast and once again read 5GB during streaming. And repeat this process for each video and each time we cast the same video.

> or even on the fly

This is currently done in extension without writing any data to disk (no delay before casting). Currently extension pipes ffmpeg output to Chromecast built-in buffer and once the video buffer is full encoding slows down (an approach I have chosen), because we do not keep old data and transcode ahead only a little, we lose seeking ability during streaming through. I can probably add audio track selection to this code without much work.

Ideally I would like to make a custom receiver app, that will support selecting track from video as it is (without any downsides), but due to my lack of time I won't be able to work on something that big right now. Would 2nd option be OK for the time being?

**Comment by chemeris** [comment]
Sure, this option should work.
